packages/tupaia-web/src/features/Map/MapOverlays/DataVisualsLayer.tsx

```diff
@@ -30,26 +34,29 @@ export const DataVisualsLayer = ({
   hiddenValues: LegendProps['hiddenValues'];
 }) => {
   const navigateToEntity = useNavigateToEntity();
-  const { serieses, measureData, entities } = useActiveMapOverlayReport();
+  const { projectCode, entityCode } = useParams();
+  const { data: entity } = useEntity(projectCode, entityCode);
+  const { serieses, measureData, activeEntity, selectedOverlay } = useMapOverlayData(hiddenValues);
+  useEffect(() => {
+    gaEvent('Map Overlays', 'Change', selectedOverlay?.name);
```

Comment:
Potentially just add a check here to make sure `selectedOverlay` is set, and make the value you're watching `selectedOverlay?.name`